_rules/error-field-is-not-identified-using-color-only

```diff
@@ -0,0 +1,107 @@
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  -  DOM Tree
+  -  CSS Styling
+acknowledgments:
+  authors:
+    -  Giacomo Petri
+---
+
+## Applicability
+
+This rule applies to any error field that is [visible][].
+
+## Expectation (1)
+
+Each test target, together with its visual context, identify the presence of errors not only through color differences.
+
+## Assumptions
+
+If the presence of errors is conveyed through the use of colors that differ not only in their hue, but that also have a difference in lightness and the difference in relative luminance between the colors leads to a contrast ratio of 3:1 or greater, then this counts as an additional visual mean.
+
+*Note*: It is possible for an error field to be distinguishable from adjacent fields but still not have a meaningful error message. In that case, it would pass Success Criterion 1.4.1: Use of Color but still fail [Success Criterion 3.3.1: Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) and, if applicable, [Scuccess Criterion 3.3.3: Error Suggestion](https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html).
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [F81 - Failure of Success Criterion 1.4.1 due to identifying required or error fields using color differences only](https://www.w3.org/WAI/WCAG21/Techniques/failures/F81.html)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The presence of errors is conveyed through text error message.
+
+```html
+<label for="first_name">First Name</label>
+<input type="text" id="first_name" autocomplete="given-name" style="border:1px solid #dd0000" aria-invalid="true" aria-describedby="first_name_error_msg">
+<p id="first_name_error_msg" class="error_msg" style="color:#dd0000">This field is empty. Enter your first name.</p>
+```
+
+#### Passed Example 2
+
+The presence of errors is conveyed also through other CSS properties:
+- font-weight: bold AND;
+- border-width: 3px.
+
+```html
+<label for="first_name" style="color:#dd0000">First Name</label>
+<input type="text" id="first_name" autocomplete="given-name" style="border:3px solid #dd0000" aria-invalid="true">
+<label for="middle_name" style="color:#333333">Middle Name (optional)</label>
+<input type="text" id="middle_name" autocomplete="additional-name" style="border:1px solid #b0b0b0" value="Michael">
+<label for="last_name" style="color:#333333">Last Name</label>
+<input type="text" id="last_name" autocomplete="family-name" style="border:1px solid #b0b0b0" value="Doe">
```

Comment:
In that example, there is no error message, while in the first one there is.
This is why I feel the Applicability needs to be made more precise. Are we targeting:
*  errored form fields (i.e., `input`, … elements as listed in [Form field label is descriptive](https://act-rules.github.io/rules/cc0f0a) for example)? Then Passed Example 1 needs to change its explication a bit (the error is indicated by a message programatically linked to the errored field);
* error messages? Then Passed Example 2 is not applicable (there is no error message)?
* [programmatic label](https://act-rules.github.io/rules/cc0f0a#programmatic-label) of errored form field? Then neither Passed Example 1 (it's a description, not a programmatic label) nor Passed Example 2 (it's the field, not the label that indicates the error);
* something else?

---
Also, related to that, what is an errored field in the first place?
I like what you do with `aria-invalid` in the example. And I think we can actually use that and define "errored field" as "with `aria-invalid` attribute value of `true` [or better different from `false` since it can also be `spelling` or `grammar`]". Plus an assumption that this is used correctly.
Then we do have an objective an unambiguous definition of which fields should be looked at. Additionally, we can likely rely on UA to set it correctly when some other mechanism says the field is invalid 🤞 

Having a clear definition of errored field, we can work on defining the visual error indicator. From examples, it seems like it should be either the field, or its programmatic label, or visibles elements used in its accessible name or description, or the element referred to by `aria-errormessage`. As far as I understand, at least one of these must have a "visual, non-hue indication".

I'm not sure we want to provide a full list of valid indications 🤔 This might be too restrictive.

---
See also what is done in [Error message describes invalid form field value](https://act-rules.github.io/rules/36b590) to work with error messages, although I'm not sure this is the best way to handle it 😕

Reply:
That's a good point I didn't think about.

>  I think we can actually use that and define "errored field" as "with aria-invalid attribute value of true [or better different from false since it can also be spelling or grammar]". Plus an assumption that this is used correctly.

In my opinion, even with assumptions about proper use of aria-invalid (which is not mandatory for compliancy), the risk to have many inapplicable implementations is high, unless we clearly state in assumptions that we assume every input with errors must have the aria-invalid="true" (or different from false) attribute.


What about replacing 
"This rule applies to any error field that is [visible][]."
with 
"This rule applies to any [visible][] input error with its context."

starting from the WCAG definition of input error

> input error
>
> information provided by the user that is not accepted
> 
> NOTE
> This includes:
> 1. Information that is required by the [Web page](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html#dfn-web-page) but omitted by the user
> 2. Information that is provided by the user but that falls outside the required data format or values

Then we need to explain what context means, but maybe we can re-use the definition of [Form Field Error Indicator](https://act-rules.github.io/rules/36b590#form-field-error-indicator); anyway, in addition to the input element already covered by definition above, it should include:
* its label (and again we can re-use something from WCAG)
    > label
    > 
    > text or other component with a text alternative that is presented to a user to identify a component within Web content
    > 
    > NOTE
    > A label is presented to all users whereas the name may be hidden and only exposed by assistive technology. In many (but not all) cases the name and the label are the same.
    > 
    > NOTE
    > The term label is not limited to the label element in HTML.
* elements visually associated to the input (e.g. error messages).